src/search.ts

```diff
@@ -16,6 +17,65 @@ function getDefaultGlobOptions(): glob.GlobOptions {
   }
 }
 
+/**
+ * If multiple paths are specific, the least common ancestor (LCA) of the search paths is used as
+ * the delimiter to control the directory structure for the artifact. This function returns the LCA
+ * when given an array of search paths
+ *
+ * Example 1: The patterns `/foo/` and `/bar/` returns `/`
+ *
+ * Example 2: The patterns `~/foo/bar/*` and `~/foo/voo/two/*` and `~/foo/mo/` returns `~/foo`
+ */
+function getMultiPathLCA(searchPaths: string[]): string {
+  if (searchPaths.length < 2) {
+    throw new Error('At least two search paths must be provided')
+  }
+
+  const commonPaths = new Array<string>()
+  const splitPaths = new Array<string[]>()
```

Comment:
I try to avoid multi-dimensional arrays but i found it the easiest to use here. We're also guaranteed beforehand that all the searchpaths are unique so the main benefit of using a set is not needed.